code/xcvm/cosmwasm/contracts/gateway/src/assets.rs

```diff
@@ -36,7 +41,12 @@ pub(crate) fn force_remove_asset(
 	deps: DepsMut<'_>,
 	asset_id: AssetId,
 ) -> std::result::Result<Response, ContractError> {
-	ASSETS.remove(deps.storage, asset_id);
-	Ok(Response::new()
-		.add_event(make_event("assets.removed").add_attribute("asset_id", asset_id.to_string())))
+	let mut resp = Response::new();
+	if let Some(asset) = ASSETS.may_load(deps.storage, asset_id)? {
+		ASSETS.remove(deps.storage, asset_id);
+		LOCAL_ASSETS.remove(deps.storage, asset.local);
```

Comment:
local is local on some chain, because same denom can be on different chains if they like to set same denom.  it is really denom + network id, in this is this is this network + denom